auacm/app/static/js/controllers/EditCompetitionController.js

```diff
@@ -0,0 +1,223 @@
+app.controller('CreateCompetitionController', ['$scope', '$http', '$location',
+        '$routeParams', function($scope, $http, $location, $routeParams) {
+    $scope.compProblems = [];
+    var competition;
+
+    if ($routeParams.cid !== undefined) {
```

Comment:
Sounds good to me. So long as we (nor any library we use) redefines `undefined`, it shouldn't be a problem.